pkg/apis/serving/v1/route_lifecycle.go

```diff
@@ -174,10 +174,13 @@ func (rs *RouteStatus) MarkCertificateReady(name string) {
 
 // MarkCertificateNotReady marks the RouteConditionCertificateProvisioned
 // condition to indicate that the Certificate is not ready.
-func (rs *RouteStatus) MarkCertificateNotReady(name string) {
+func (rs *RouteStatus) MarkCertificateNotReady(c *v1alpha1.Certificate) {
```

Comment:
So `NotReady` doesn't imply 'failed' - so at this point we the certificate might still be issuing.

To fix the 'failed' case with a status message we should update the method `MarkCertificateProvisionFailed`

Reply:
It seems like the reconciler doesn't `MarkCertificateProvisionFailed` when a kcert is created even when the certificate resource from cert-manager is never created. The [code](https://github.com/knative/serving/blob/ecbe40a5feae924fcd20fb77e5a7b04003a05fb6/pkg/reconciler/route/route.go#L229) currently looks at the presence of the kcert resource and not look at its status and hence it stays in `Not ready` state. May be this is also something that can be addressed as marking certificate provision failure makes sense in this case. May be this can be addressed in a separate issue. I can close this PR and create a new issue and mark this as a dependant. What do you think @dprotaso?

Reply:
> May be this is also something that can be addressed as marking certificate provision failure makes sense in this case. May be this can be addressed in a separate issue.

Sounds good let's consider that a separate issue


> I can close this PR and create a new issue and mark this as a dependant. 
I still think there's value is propagating the (not ready) unknown message - so let's keep this PR

Also might be worth seeing how we propagate status from Route => Service

https://github.com/knative/serving/blob/5b1eab3762be7040e3c2d348d74bb72f1dcc3890/pkg/apis/serving/v1/service_lifecycle.go#L130